lld/MachO/MapFile.cpp

```diff
@@ -203,6 +205,8 @@ void macho::writeMapFile() {
     for (const OutputSection *osec : seg->getSections()) {
       if (auto *concatOsec = dyn_cast<ConcatOutputSection>(osec)) {
         for (const InputSection *isec : concatOsec->inputs) {
+          if (isec->isLinkOptimizedAway())
+            continue;
```

Comment:
can we filter these out in `gatherInputSections` instead of doing it here?

(though as per comment on the gdoc, I would prefer if we just used the existing ICF/live flags since there should already be logic for handling those

Reply:
Will switch to just using 'live'.